test/Microsoft.IdentityModel.JsonWebTokens.Tests/JsonWebTokenHandlerTests.cs

```diff
@@ -2707,6 +2793,49 @@ public class CreateTokenTheoryData : TheoryDataBase
 
         public IEnumerable<SecurityKey> ExpectedDecryptionKeys { get; set; }
     }
+
+    // Overrides CryptoProviderFactory.CreateAuthenticatedEncryptionProvider to create AuthenticatedEncryptionProviderMock that provides AesGcm encryption.
+    public class CryptoProviderFactoryMock: CryptoProviderFactory
+    {
+        public override AuthenticatedEncryptionProvider CreateAuthenticatedEncryptionProvider(SecurityKey key, string algorithm)
+        {
+            if (SupportedAlgorithms.IsSupportedEncryptionAlgorithm(algorithm, key) && SupportedAlgorithms.IsAesGcm(algorithm))
+                return new AuthenticatedEncryptionProviderMock(key, algorithm);
+
+            return null;
+        }
+    }
+
+    // Overrides AuthenticatedEncryptionProvider.Encrypt to offer AesGcm encryption for testing.
+    public class AuthenticatedEncryptionProviderMock: AuthenticatedEncryptionProvider
+    {
+        public AuthenticatedEncryptionProviderMock(SecurityKey key, string algorithm): base(key, algorithm)
+        { }
+
+        public override AuthenticatedEncryptionResult Encrypt(byte[] plaintext, byte[] authenticatedData)
+        {
+            byte[] nonce = new byte[AesGcm.NonceSize];
+
+            // Generate secure nonce
```

Comment:
nit: consider removing the word 'secure' from this comment, perhaps replace it with 'test' or similar.

Reply:
Fixed it